File: Programming_Basics_with_Python/conditional_statements_advanced _lab/12._trade_commissions.py
name_of_city = input()
sales = float(input())
commissions = 0
if not name_of_city in ("Sofia, Varna, Plovdiv"):
    print("error")
    exit()
if sales < 0:
    print("error")
    exit()
if name_of_city == "Sofia":
    if 0 <= sales <= 500:
        commissions = sales * 0.05
    elif 500 < sales <= 1000:
        commissions = sales * 0.07
    elif 1000 < sales <= 10000:
        commissions = sales * 0.08
    elif sales > 10000:
        commissions = sales * 0.12
    print(f"{commissions:.2f}")
elif name_of_city == "Varna":
    if 0 <= sales <= 500:
        commissions = sales * 0.045
    elif 500 < sales <= 1000:
        commissions = sales * 0.075
    elif 1000 < sales <= 10000:
        commissions = sales * 0.10
    elif sales > 10000:
        commissions = sales * 0.13
    print(f"{commissions:.2f}")
elif name_of_city == "Plovdiv":
    if 0 <= sales <= 500:
        commissions = sales * 0.055
    elif 500 < sales <= 1000:
        commissions = sales * 0.08
    elif 1000 < sales <= 10000:
        commissions = sales * 0.12
    elif sales > 10000:
        commissions = sales * 0.145
    print(f"{commissions:.2f}")

File: Programing_Basics_with_JS/conditionalStatementsAdvancedLab/12.TradeCommissions.js
function tradeCommissions(input) {
    let city = input[0];
    let sales = Number(input[1]);

    let commissions = 0;

    switch(city) {
        case 'Sofia':
            if (0 <= sales && sales <= 500) {
                commissions = sales * 5 / 100;
            } else if (500 < sales && sales <= 1000) {
                commissions = sales * 7 / 100;
            } else if (1000 < sales && sales <= 10000) {
                commissions = sales * 8 / 100;
            } else if (10000 < sales) {
                commissions = sales * 12 / 100;
            } else console.log('error')
            break;
        case 'Varna':
            if (0 <= sales && sales <= 500) {
                commissions = sales * 4.5 / 100;
            } else if (500 < sales && sales <= 1000) {
                commissions = sales * 7.5 / 100;
            } else if (1000 < sales && sales <= 10000) {
                commissions = sales * 10 / 100;
            } else if (10000 < sales) {
                commissions = sales * 13 / 100;
            } else console.log('error')
            break;
        case 'Plovdiv':
            if (0 <= sales && sales <= 500) {
                commissions = sales * 5.5 / 100;
            } else if (500 < sales && sales <= 1000) {
                commissions = sales * 8 / 100;
            } else if (1000 < sales && sales <= 10000) {
                commissions = sales * 12 / 100;
            } else if (10000 < sales) {
                commissions = sales * 14.5 / 100;
            } else console.log('error')
            break;
        default: console.log('error')
    }

    if (commissions > 0) {
        console.log(commissions.toFixed(2));
    }  
}

tradeCommissions(['Sofia', '300'])
tradeCommissions(['Plovdiv', '499.99'])
tradeCommissions(['Varna', '3874.50'])
tradeCommissions(['Kaspichan', '300'])
tradeCommissions(['Sofia', -'50'])
